### PR TITLE
Dvd iso volume size

### PR DESCRIPTION
**Pull request by superg**
WIP

## Summary by CodeRabbit

* **New Features**
  * Unified DVD dumping with improved Xbox/Kreon handling, protection-aware skip/write behavior, and new Xbox security/context utilities.
  * New array-based endian conversion helpers.

* **Bug Fixes**
  * Stronger range handling with automatic merging and stricter validation/error behavior.
  * Improved read/write error reporting, retries and fallback behavior; refined security-sector handling.

* **Tests**
  * Added comprehensive range tests covering merges, boundaries, and edge cases.

* **Chores**
  * Updated drive metadata, generalized range/string parsing, module reorganization, and increased CI verbosity.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: drive.ixx [hunk 1]
```diff
@@ -127,86 +127,88 @@ static const DriveConfig DRIVE_CONFIG_GENERIC = { "", "", "", "", 0, 0, 0, ReadM
...
+    { "BUFFALO" , "Optical Drive"    , "1.00", "N003103MO6LAL95700"  ,   +6,   0, -135, ReadMethod::BE, SectorOrder::DATA_C2_SUB, Type::LG_ASU8B }, // Low_Plankton_3329
     // PATCHED
-    { "ASUS"    , "BW-16D1HT"        , "3.10", "WM01601KLZL4TG5625" ,   +6,   0, -135, ReadMethod::BE, SectorOrder::DATA_C2_SUB, Type::GENERIC  }, // 3.10MK or RibShark FW definition
-    { "TSSTcorp", "DVD-ROM TS-H353C" , "ZZ00", ""                   ,   +6,   0, -135, ReadMethod::BE, SectorOrder::DATA_C2_SUB, Type::GENERIC  }, // MoriGM
+    { "ASUS"    , "BW-16D1HT"        , "3.10", "WM01601KLZL4TG5625"  ,   +6,   0, -135, ReadMethod::BE, SectorOrder::DATA_C2_SUB, Type::GENERIC  }, // 3.10MK or RibShark FW definition
+    { "TSSTcorp", "DVD-ROM TS-H353C" , "ZZ00", ""                    ,   +6,   0, -135, ReadMethod::BE, SectorOrder::DATA_C2_SUB, Type::GENERIC  }, // MoriGM
+    // PATCHED KREON
+    { "TSSTcorp", "DVD-ROM SH-D163B" , "ZZ01", "KREON V1.00.........",   +6,   0, -135, ReadMethod::BE, SectorOrder::DATA_C2_SUB, Type::GENERIC  }, // MoriGM
```
Comment (Deterous) [inline drive.ixx]: D163B is a generic / bad drive for CD dumping, isn't it?

File: cd/cd_dump.ixx [hunk 2]
```diff
@@ -276,14 +276,13 @@ export bool redumper_dump_cd(Context &ctx, const Options &options, DumpMode dump
 
     std::vector<Range<int32_t>> protection;
     protection_to_ranges(protection, ctx.protection);
-    protection_ranges_from_lba_ranges(protection, string_to_ranges(options.skip), -ctx.drive_config.read_offset);
+    protection_ranges_from_lba_ranges(protection, string_to_ranges<int32_t>(options.skip), -ctx.drive_config.read_offset);
```
[the remark this answers is not in the document]
Comment (superg) [reply cd/cd_dump.ixx]: so rvalue is passed to protection_from_lba_ranges() as std::span, internally function iterates over this span and fills protection vector. Isn't rvalue validity guaranteed during protection_ranges_from_lba_ranges() execution?

File: dvd/dvd_dump.ixx [hunk 3]
```diff
@@ -375,34 +377,40 @@ export bool redumper_dump_dvd(Context &ctx, const Options &options, DumpMode dum
...
+    uint32_t sectors_count_capacity;
+    {
+        uint32_t sector_last, block_length;
+        auto status = cmd_read_capacity(*ctx.sptd, sector_last, block_length, false, 0, false);
+        if(status.status_code)
+            throw_line("failed to read capacity, SCSI ({})", SPTD::StatusMessage(status));
+        if(block_length != FORM1_DATA_SIZE)
+            throw_line("unsupported block size (block size: {})", block_length);
+        sectors_count_capacity = sector_last + 1;
+    }
```
[the remark this answers is not in the document]
Comment (superg) [reply dvd/dvd_dump.ixx]: The current code is correct, no need to change anything.
The issue is that if I use sectors_count_capacity, it causes under dumping.
The intent here is to centralize sector count logic in one place because there will be more opportunity to choose from, ISO/UDF filesystem size, I'll have such options, that's why this code block is like that.
